File: 330-patching-array/patching-array.c
#include <stdio.h>

int minPatches(int* nums, int numsSize, int n) {
    long long patch = 1;
    int total = 0;
    int index = 0;

    while (patch <= n) {
        if (index < numsSize && patch >= nums[index]) {
            patch += nums[index];
            index++;
        } else {
            patch += patch;
            total++;
        }
    }

    return total;
}

File: 330-patching-array/patching-array.py
class Solution:
    def minPatches(self, nums, n):
        patch = 1
        total = 0
        index = 0

        while patch <= n:
            if index < len(nums) and patch >= nums[index]:
                patch += nums[index]
                index += 1
            else:
                patch += patch
                total += 1

        return total
